python/ray/tests/spark/test_databricks_hook.py

```diff
@@ -0,0 +1,81 @@
+import sys
+
+import mock
+import pytest
+import os
+import time
+import ray
+from pyspark.sql import SparkSession
+from ray.util.spark import setup_ray_cluster
+
+
+pytestmark = pytest.mark.skipif(
+    not sys.platform.startswith("linux"),
+    reason="Ray on spark only supports running on Linux.",
+)
+
+
+class MockDbApiEntry:
+
+    def __init__(self):
+        self.idle_time = 0
+        self.registered_job_groups = []
+
+    def getIdleTimeMillisSinceLastNotebookExecution(self):
+        return (time.time() - self.idle_time) * 1000
+
+    def registerBackgroundSparkJobGroup(self, job_group_id):
+        self.registered_job_groups.append(job_group_id)
+
+
+class TestDatabricksHook:
+    @classmethod
+    def setup_class(cls):
+        os.environ["SPARK_WORKER_CORES"] = "2"
```

Comment:
Mutating `os.environ` like this affects other tests should be avoided or you can use pytest's `monkeypatch.setenv`

Reply:
btw is there a reason you're using unittest's style rather than pytest's style?